OpenRA.Mods.Common/Traits/World/Pathfinder/Constants.cs

```diff
@@ -0,0 +1,28 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public static class Constants
+	{
+		/// <summary>
+		/// We can't rely on floating point math to be deterministic across all runtimes.
+		/// The cases that use this will need to be changed to use integer math
+		/// </summary>
+		public const double Sqrt2 = 1.414;
```

Comment:
The short answer is "Probably. Don't risk it".  Desyncs used to be a serious problem in earlier versions of the engine, and many of these could be explicitly traced back to disagreements in floating point calculations.  We only finally killed them off by mass-rewriting the simulation logic to use integer math or the (software-emulated, therefore slow) decimal type.

I'm not an expert, but my understanding is that the disagreements creep in when converting the (possibly extended precision, depending on your FPU configuration) result of a native fp calculation back to a .NET/mono float/double.  I'm not sure whether this is due to bugs in mono or an explicit grey area in the spec, but in any case it is a real problem that we have to deal with.  Rounding an 80bit float down to 64bit is less likely to have issues than going to 32, but they can't be ruled out.  It's probably possible to write desync-safe fp code if you're careful about casting intermediate results, but this makes for fragile code that people are (or should be) afraid to touch.